src/output.c

```diff
@@ -1034,6 +1025,149 @@ void OutputRegisterRootLoggers(void)
     OutputStreamingLoggerRegister();
 }
 
+static int JsonGenericLogger(ThreadVars *tv, void *thread_data, const Packet *p, Flow *f,
+        void *state, void *tx, uint64_t tx_id, int dir)
+{
+    OutputJsonThreadCtx *thread = thread_data;
+    EveJsonSimpleAppLayerLogger *al = SCEveJsonSimpleGetLogger(f->alproto);
+    if (al == NULL) {
+        return TM_ECODE_FAILED;
+    }
+
+    const char *name;
+    switch (al->proto) {
+        case ALPROTO_HTTP2:
+            // special case
+            name = "http";
+            break;
+        case ALPROTO_FTPDATA:
+            // underscore instead of dash
+            name = "ftp_data";
+            break;
+        case ALPROTO_BITTORRENT_DHT:
+            // underscore instead of dash
+            name = "bittorrent_dht";
+            break;
+        default:
+            name = AppProtoToString(al->proto);
+    }
+    JsonBuilder *js = CreateEveHeader(p, dir, name, NULL, thread->ctx);
+    if (unlikely(js == NULL)) {
+        return TM_ECODE_FAILED;
+    }
+
+    if (!al->LogTx(tx, js)) {
+        goto error;
+    }
+
+    OutputJsonBuilderBuffer(js, thread);
+    jb_free(js);
+
+    return TM_ECODE_OK;
+
+error:
+    jb_free(js);
+    return TM_ECODE_FAILED;
+}
+
+static int JsonGenericDirPacketLogger(ThreadVars *tv, void *thread_data, const Packet *p, Flow *f,
```

Comment:
I like the removal of files, but I don't like how we now have all these protocol specific functions here. It feels like a step backwards. What is the purpose of it? Do have have some next step in mind where it can be registered from the parsers directly?

Reply:
Do you mean the ones like `OutputSIPLogInitSub` ?

This `OutputSIPLogInitSub` currently lives in output-json-sip.c on the master branch.
Would you like to keep output-json-sip.c just to have this `OutputSIPLogInitSub` ?
Or would you like the calls to `AppLayerParserRegisterLogger` to be moved in rust/src/sip ?

Reply:
Just pushed a draft commit on top to show the second possibility

Reply:
I do like the second possibility more.  I'd rather see less JSON stuff in `output.c` than more, as they *should* be at different levels of abstraction that got a little messy at some point.

Reply:
Ok doing in next PR